app/components/github-repo.js

```diff
@@ -3,20 +3,46 @@ import recordsList from 'code-corps-ember/utils/records-list';
 
 const {
   computed,
+  computed: { alias, filter, mapBy, notEmpty },
   Component,
-  get
+  get,
+  inject: { service }
 } = Ember;
 
 export default Component.extend({
   classNames: ['github-repo'],
+  classNameBindings: ['isConnected:github-repo--connected'],
   tagName: 'li',
 
-  isConnected: computed('githubRepo', 'project.projectGithubRepos.@each', function() {
-    let githubRepo = get(this, 'githubRepo');
-    let project = get(this, 'project');
-    let projectGithubRepos = get(this, 'project.projectGithubRepos');
-    if (projectGithubRepos) {
-      return recordsList.includes(projectGithubRepos, githubRepo);
+  githubRepo: null,
+  project: null,
+
+  store: service(),
+
+  isConnected: notEmpty('projectsGithubRepo'),
+  projectGithubRepo: computed('projectGithubRepos.@each', 'projectGithubRepos.isFulfilled', 'githubRepo.id', function() {
```

Comment:
Either way, the naming here is not at all clear to me and it seems like maybe this is a case of overextending concerns, as in this component cares too much about its parent/siblings. Maybe the parent (controller?) should just map out a collection and assign an `isConnected` property to the component?

Again, not sure. Just trying to find an alternative to something that keeps being problematic for us.